app/api/fiat-currency-price/route.ts

```diff
@@ -0,0 +1,31 @@
+import { coingeckoClient } from "@/services/coingecko/client"
+
+export const revalidate = 60 * 10
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url)
+  const id = searchParams.get("id")
+  const currency = searchParams.get("currency")
+
+  try {
+    if (!id || !currency) {
+      throw new Error("Invalid request")
+    }
+
+    const res = await coingeckoClient.get(
+      `/simple/price?ids=${id}&vs_currencies=${currency}`
+    )
+
+    const currentFiatPrice = res.data
+
+    return new Response(JSON.stringify({ currentFiatPrice }), {
+      headers: {
+        "content-type": "application/json",
+      },
+    })
+  } catch (e: any) {
+    return new Response(e.message, {
+      status: 400,
+    })
+  }
+}
```

Comment:
Why not use req.query and res.json? THis all seems overly complicated
https://nextjs.org/docs/pages/building-your-application/routing/api-routes

Reply:
I am not opposed to using the "old syntax" (pages directory) but the one from the app directory is the documentation that users will likely refer to. I have simplified my code to streamline and use the same syntax as in this documentation https://nextjs.org/docs/app/building-your-application/routing/route-handlers